google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageReadChannel.java

```diff
@@ -41,27 +42,42 @@
 @ThreadSafe
 final class CloudStorageReadChannel implements SeekableByteChannel {
 
-  private final ReadChannel channel;
+  private final Storage gcsStorage;
+  private final BlobId file;
+  // max # of times we may reopen the file
+  private final int maxReopen;
+  // how many times we re-opened the file
+  private int reopens;
+  private ReadChannel channel;
   private long position;
   private long size;
 
+  /**
+   * @param maxReopen max. number of times to try re-opening the channel if it closes on us unexpectedly.
```

Comment:
the "." after "max" will result in javadoc summary line containing only the word "max". I recommend not using the ".".